invenio_records_rest/views.py

```diff
@@ -0,0 +1,239 @@
+        self.resolver = resolver
+        self.minter = current_pidstore.minters[minter_name]
+
+    def post(self, **kwargs):
+        """Create a record.
+
+        :returns: The created record.
+        """
+        if request.content_type != 'application/json':
+            abort(415)
+
+        # TODO: accept non json content (MARC21...)
+        data = request.get_json()
+        if data is None:
+            return abort(400)
+
+        try:
+            # Create uuid for record
+            record_uuid = uuid.uuid4()
+            # Create persistent identifier
+            pid = self.minter(record_uuid, data=data)
+            # Create record
+            record = Record.create(data, id_=record_uuid)
+
+            db.session.commit()
+        except SQLAlchemyError:
+            db.session.rollback()
+            raise
+        return self.make_response(pid, record, 201)
+
+
+class RecordResource(ContentNegotiatedMethodView):
+    """Resource for record items."""
+
+    view_name = '{0}_item'
+
+    def __init__(self, resolver=None, **kwargs):
+        """Constructor.
+
+        :param resolver: Persistent identifier resolver instance.
+        """
+        super(RecordResource, self).__init__(**kwargs)
+        self.resolver = resolver
+
+    @pass_record
+    def get(self, pid, record, **kwargs):
+        """Get a record.
+
+        :param pid: Persistent identifier for record.
+        :param record: Record object.
+        :returns: The requested record.
+        """
+        self.check_etag(str(record.model.version_id))
+        return pid, record
+
+    @require_content_types('application/json-patch+json')
+    @pass_record
+    def patch(self, pid, record, **kwargs):
+        """Modify a record.
+
+        The data should be a JSON-patch, which will be applied to the record.
+
+        :param pid: Persistent identifier for record.
+        :param record: Record object.
+        :returns: The modified record.
+        """
+        # TODO: accept 'application/json' mediatype and use the object
+        # to replace the specified attributes
+        data = request.get_json(force=True)
+        if data is None:
+            abort(400)
+
+        self.check_etag(str(record.model.version_id))
+        try:
+            record = record.patch(data)
+        except (JsonPatchException, JsonPointerException):
+            abort(400)
+
+        record.commit()
+        db.session.commit()
+        return pid, record
+
+    @require_content_types('application/json')
+    @pass_record
+    def put(self, pid, record, **kwargs):
+        """Replace a record.
+
+        The body should be a JSON object, which will fully replace the current
+        record metadata.
+
+        :param pid: Persistent identifier for record.
+        :param record: Record object.
+        :returns: The modified record.
+        """
+        # TODO: accept non json content (MARC21...)
+        data = request.get_json()
```

Comment:
request.get_json doc says: `By default this function will only load the json data if the mimetype is application/json`. Just to be sure I tested with `application/json-patch+json`. It fails.
What about doing this:
* if the content type is `application/json-patch+json` and call `request.get_json(froce=True)` then `record.patch(data)`.
* else call `request.get_json()`, if the result is None and content-type is `application/json` or not set at all, fail with 400, else fail with 415